app/src/main/java/com/zulip/android/ExpandableStreamDrawerAdapter.java

```diff
@@ -0,0 +1,41 @@
+package com.zulip.android;
+
+import android.content.Context;
+import android.database.Cursor;
+import android.database.MatrixCursor;
+import android.widget.SimpleCursorTreeAdapter;
+
+import com.zulip.android.models.Message;
+
+import java.util.List;
+
+public class ExpandableStreamDrawerAdapter extends SimpleCursorTreeAdapter {
+
+    public ExpandableStreamDrawerAdapter(final Context context, Cursor cursor, int groupLayout,
+                                         String[] groupFrom, int[] groupTo, int childLayout,
+                                         String[] childFrom, int[] childTo) {
+        super(context, cursor, groupLayout, groupFrom, groupTo, childLayout, childFrom, childTo);
+    }
+
+    @Override
+    public Cursor getChildrenCursor(Cursor groupCursor) {
+        List<String[]> results = null;
+        try {
+            results = ZulipApp.get().getDao(Message.class).queryRaw("SELECT DISTINCT subject FROM messages " +
+                    "JOIN streams ON streams.id=messages.stream " +
+                    "WHERE streams.id=" + groupCursor.getInt(0)).getResults();
+        } catch (Exception e) {
```

Comment:
it's a good idea to be explicit about what types of exceptions you're catching.
 Add a line note